simphony_ui/openfoam_boundary_conditions.py

```diff
@@ -0,0 +1,98 @@
+from traits.api import HasStrictTraits, Float, Enum, Instance, Array
+from traitsui.api import View, Item, Tabbed, UItem
+
+
+class VelocityBoundaryConditionModel(HasStrictTraits):
+
+    type = Enum('none', 'fixedGradient', 'fixedValue')
+
+    fixed_gradient = Array(Float, (3,))
+
+    fixed_value = Array(Float, (3,))
+
+    traits_view = View(
+        'type',
+        Item(
+            name='fixed_gradient',
+            visible_when='type == "fixedGradient"',
+            style='custom'
+        ),
+        Item(
+            name='fixed_value',
+            visible_when='type == "fixedValue"',
+            style='custom'
+        ),
+    )
+
+
+class PressureBoundaryConditionModel(HasStrictTraits):
```

Comment:
These two are the same. make a base class and inherit from each, so that you have different types. No need to duplicate the declaration.